Mapping/inc/Obstacles.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Header file providing class for obstacles
+ * Author: Frederick Wachter
+ */
+
+#ifndef __OBSTACLES__
+#define __OBSTACLES__
+
+#include <array>
+using namespace std;
+
+class Obstacles {
+	public:
+		bool *directions[4] = { &north, &east, &south, &west };
```

Comment:
shouldn't north, east, south, and west be declared before this. You can't grab a memory reference to an object that has not been declared yet. It may be better to do this in the constructor.